### PR TITLE
openstack-tests: Add to installer presubmits

### PR DESCRIPTION
**Pull request by pierreprinetti**
This patch adds an openstack-tests step to the e2e-openstack jobs among
the Installer presubmits.

Implements [OSASINFRA-2682](https://issues.redhat.com/browse/OSASINFRA-2682)

### CODE REVIEW

File: ci-operator/config/openshift-priv/installer/openshift-priv-installer-master.yaml
Comment (mandre) [inline ci-operator/config/openshift-priv/installer/openshift-priv-installer-master.yaml:42]: Leave openshift-priv changes outside of this PR. It's going to be updated automatically via a bot after your PR merges.

https://docs.ci.openshift.org/docs/architecture/private-repositories/#openshift-priv-organization

**Review by mandre — COMMENTED** [review]
Note to reviewers: this adds the openstack-tests image definition everywhere we have an openstack job running. This also points to openstack-tests:4.10 in all branches, as this is the only image we build.

/lgtm

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml
Comment (vrutkovs) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml:7]: This either needs to match "latest" release there (4.9) or should be removed if the change was not backported there

Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml:7]: OK here's my plan:
1. Add a build trigger for creating images for all releases: https://github.com/openshift/release/pull/23629
2. Create the actual release branches on the openstack-tests repo
3. change this PR to include images for all respective branches

Do you think this plan makes sense?

Comment (vrutkovs) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml:7]: yeah, that looks good. Perhaps we'd get feedback earlier if we focus on 4.10 for now - and later do the backports and other updates?

Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml:7]: >if we focus on 4.10 for now

By "focus on 4.10 now", do you mean:
A) merging this as-is first; OR
B) only adding the image to 4.10 jobs

If A), I'd be fine with it.

If B): The goal here is to add an `openstack-test` to every `e2e-openstack` workflow. Which in turn, applies to every release. Thus _I think_ we need to add a valid image to every version first.

Comment (vrutkovs) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml:7]: I think I'd prefer B) first, as in any case we'd have to create subsequent PRs to add this stanza or update tag from 4.10 everywhere - and repeated PRs to add stanza to 4.9/4.8 would look a bit more consistent. However the difference is negligible really (we should link to this discussion in follow up PRs)

**Comment by vrutkovs** [comment]
/approve

Needs a rebase due to conflict

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml
Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml]: I have repurposed this PR in an attempt to respond to the concerns you raise.

This PR is now adding the image to `installer/master` only, and chaging one single job to include the relevant step. It also adds the step in the step-registry.

Is this actually a proper solution to the concerns you raised, @vrutkovs ?

**Comment by pierreprinetti** [comment]
Success!
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/23623/rehearse-23623-pull-ci-openshift-installer-master-e2e-openstack/1460643073687031808

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml
Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml]: The test with 4.10 worked well. I am then completing the work for the whole Installer repo.

Comment (pierreprinetti) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.9-upgrade-from-stable-4.8.yaml]: The rehearsal with Installer master worked well.
One way forward could be to merge this and open a new PR applying the same changes everywhere.

What do you suggest, @vrutkovs?

**Review by mandre — COMMENTED** [review]
It works!
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/23623/rehearse-23623-pull-ci-openshift-installer-master-e2e-openstack/1461312337381167104/artifacts/e2e-openstack/openstack-test-openstack/build-log.txt

/lgtm

**Comment by pierreprinetti** [comment]
Since:
* the rehearsals on the master branch worked so far
* approvals are hard to get
* Installer is apparently a high-traffic repo (I had to rebase multiple times in a few days)

I am extending the change to all Installer branches, so that when we get the approval, we won't have to start over again for each branch.

I am not touching any other repository, because those may require different approvers.

**Review by vrutkovs — APPROVED** [review]
/lgtm

**Comment by pierreprinetti** [comment]
Removed the `openstack-tests` image from the 4.5 config, as it was not found. There are no step-registry openstack tests in 4.5 anyway

**Comment by pierreprinetti** [comment]
/hold
we're back in the realm of debugging

**Comment by pierreprinetti** [comment]
testplatform has triggered imagebuild manually; rehearsals should eventually succeed for 4.9
/retest

**Comment by pierreprinetti** [comment]
depends on https://github.com/openshift/release/pull/23805

**Comment by pierreprinetti** [comment]
/hold cancel
I don't know what is going on with 4.7, but it should not be an issue. If ever that problem arises in an actual PR, we will be able to debug there. The problem arises AFTER the Conformance run, which is independently trackable in the test results.

I deem this is ready to merge.

**Review by mandre — COMMENTED** [review]
/lgtm

**Comment by pierreprinetti** [comment]
/close
in favour of smaller steps.

**Pull request closed without merging**
